### PR TITLE
Update to Azure SAML setup

### PR DESCRIPTION
**Pull request by tofuSCHNITZEL**
I just managed to setup AzureAD SAML auth with kimai and I recommend some improvments to the Azure saml setup documentation:

- I would suggest to remove
`- { saml: $http://schemas.xmlsoap.org/ws/2005/05/identity/claims/employeeid, kimai: accountNumber }`
since "employeeid" does not exists by default in azure ad (I had php 500 errors with employeeid "Missing user attribute")

 - I suggest adding 
 ```
 security:
                 requestedAuthnContext: false
 ```
 so that the login also works with edge browser where users are already signed in to Microsoft 365 via the Windows settings and don't (can't) use a password
 
- on step two I suggest to remove singleLogoutService since its not present in the example config above and I also don't recommend using it because on kimai logout you normally don't want to get logged out of Microsoft 365 as well
 
- updated config at the steps to be consistent with the default config from above
 
- added some clarification to the x509cert config step

### CODE REVIEW

File: _documentation/saml-azure.md [hunk 1]
```diff
@@ -96,25 +97,18 @@ title: Login with Azure
...
 ```yaml
 entityId: 'https://timetracking.example.com/auth/saml/metadata'
 assertionConsumerService:
     url: 'https://timetracking.example.com/auth/saml/acs'
-    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST'
-# logout does not seem to work properly
-#singleLogoutService:
-#    url: 'https://timetracking.example.com/auth/saml/logout'
-#    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
+singleLogoutService:
```
Comment (kevinpapst) [inline _documentation/saml-azure.md:107]: > on step two I suggest to remove singleLogoutService since its not present in the example config above and I also don't recommend using it because on kimai logout you normally don't want to get logged out of Microsoft 365 as well

Why did you remove the comment then?

File: _documentation/saml-azure.md [hunk 2]
```diff
@@ -84,6 +83,8 @@ kimai:
                     url: 'https://timetracking.example.com/auth/saml/acs'
                 singleLogoutService:
                     url: 'https://timetracking.example.com/auth/saml/logout'
+            security:
+                requestedAuthnContext: false
```
Comment (kevinpapst) [inline _documentation/saml-azure.md:87]: Do you have some documentation on that? I do not have access to a Windows machine for testing.

**Review by kevinpapst — COMMENTED** [review]
Thanks for your input! Can you have a look at my two comments please

File: _documentation/saml-azure.md (hunk 1 above)
Comment (tofuSCHNITZEL) [reply _documentation/saml-azure.md:107]: sorry should have been clearer  - I removed the singleLogoutService from the **IDP** part but I left it in the **SP** part because the comment said "seems to not work" but in fact it was not working for IDP but I see no errors with my config where singleLogoutService  is present for SP

File: _documentation/saml-azure.md (hunk 2 above)
Comment (tofuSCHNITZEL) [reply _documentation/saml-azure.md:87]: I tried various variation of this setting on my machine. setting it to "true" (the default) gives this error when logging in with edge:
`AADSTS75011: Authentication method 'X509, MultiFactor' by which the user authenticated with the service doesn't match requested authentication method 'Password'. Contact the Kimai application owner.`

unfortunately setting it to an array like suggested here: https://github.com/SAML-Toolkits/php-saml#how-it-works

requestedAuthnContext: 
   - 'urn:oasis:names:tc:SAML:2.0:ac:classes:X509'
   - 'urn:oasis:names:tc:SAML:2.0:ac:classes:Password'

makes it work in edge but then you get a similar error like above when logging in with password... 

see also here:
https://learn.microsoft.com/en-us/troubleshoot/azure/active-directory/error-code-aadsts75011-auth-method-mismatch

Comment (kevinpapst) [reply _documentation/saml-azure.md:87]: Super interesting. I had a user with an issue that sounds similar, which I was unable to solve: 
https://github.com/kimai/kimai/issues/3698

I wonder if removing that has any security implications. But it seems, it only configures which authentication methods are allowed. And that is really not up to Kimai to decide, but the job of the SAML application owner.

That is a great finding, thank you @tofuSCHNITZEL 👍

Comment (tofuSCHNITZEL) [reply _documentation/saml-azure.md:87]: yes it will fix the problem in https://github.com/kimai/kimai/issues/3698

File: _documentation/saml-azure.md (hunk 1 above)
Comment (kevinpapst) [reply _documentation/saml-azure.md:107]: Ah right, now I see the change. Diffs are sometimes weird to read...

You removed the `binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'` for the `singleSignOnService` - did you try that it still works?

**Review by kevinpapst — APPROVED** [review]

**Comment by kevinpapst** [comment]
Took a second to reactivate my test environment... I tried everything that I asked and it worked. Awesome.
Thank you, that one line with `requestedAuthnContext: false` was a fantastic finding 🎉

File: _documentation/saml-azure.md (hunk 1 above)
Comment (tofuSCHNITZEL) [reply _documentation/saml-azure.md:107]: yes I removed the binding since it worked also without it..

**Comment by tofuSCHNITZEL** [comment]
sorry to come back to this... but I just realised something:
with:
```
            mapping:
                - { saml: xxxxxxxx-yyyy-xxxx-yyyy-xxxxxxxxxxxx, kimai: ROLE_ADMIN }
                - { saml: Azure-Group-Object-Id, kimai: ROLE_TEAMLEAD }
```
should "xxxxxxxx-yyyy-xxxx-yyyy-xxxxxxxxxxxx" be replaced with something concrete from the azureAd environment? (e.g. the uuid of the global admin group of the azuread tenant?)

**Comment by kevinpapst** [comment]
???
Yes. 
Obviously `xxxxxxxx-yyyy-xxxx-yyyy-xxxxxxxxxxxx` is just a placeholder.

**Comment by kevinpapst** [comment]
Hm, what could `Azure-Group-Object-Id` mean?

image

**Comment by tofuSCHNITZEL** [comment]
sure now that you clarified it, its obvious...

**Comment by kevinpapst** [comment]
I pushed a docu update with an example. Thanks for pointing it out 👍

**Comment by tofuSCHNITZEL** [comment]
thanks for adding the group id mapping explaination to the documentation!
